src/eval-frame/iodide-api/file.js

```diff
@@ -0,0 +1,96 @@
+import { getState as reduxStoreGetState } from "../store";
+import { getFiles } from "../../editor/tools/server-tools";
+import sendFileRequestToEditor from "../tools/send-file-request-to-editor";
+import { FETCH_RETURN_TYPES } from "../../editor/state-schemas/state-schema";
+
+const DEFAULT_SAVE_OPTIONS = { overwrite: false };
+
+function isString(argument) {
+  return typeof argument === "string" || argument instanceof String;
+}
+
+function confirmIsString(key, argument) {
```

Comment:
i really like that you are being aggressive about checking argument types and throwing when they are bad -- this will help users catch errors early, and i think we should implement it for all of our iodide API functions. it's such a good idea, in fact, that i think you should abstract it out into something re-usable in it's own file `validate-arguments` or something, and that we should make the apia bit more generic, maybe inspiresd by proptypes or something. could be like:

```
export function loadFile(fileName, fetchType, variableName = undefined) {
  argType.validate({
    fileName: [fileName, argType.string],
    fetchType: [fetchType, argType.enum(FETCH_RETURN_TYPES)],
    variableName: [variableName, argType.string.optional]
  })
  return sendFileRequestToEditor(fileName, "LOAD_FILE", {
    fetchType
  }).request.then(handleResourceLoad(fetchType, variableName));
}
```

no need to add any additional validation functions right now, but this is a great idea and i think we should go ahead and make a foundation we can build on. (and it occurs to me that there may be an existing lib that does this better than that strawman API, worth a quick google search)

Reply:
At this point, this is the only bit of feedback I haven't implemented yet. I'd like to think about this a bit. On the one hand, I like the idea of using something for these types of validations, but on the other, I don't want to reinvent the wheel too much. Either way, given that the API surface of this PR is actually fairly small (save, load, delete, exists, lastUpdated, list) this might be more worthwhile to tackle in a follow-on PR.

Reply:
agreed, this is not critical. filed #1760